DPL: close (almost) all file descriptors before forking

ktf opened the pull request.

ktf committed:
DPL: close (almost) all file descriptors before forking

ktf commented:
@chiarazampolli I think this is it.

davidrohr commented on Framework/Core/src/runDataProcessing.cxx:
```diff
@@ -726,20 +726,29 @@ void spawnDevice(DeviceRef ref,
+      }
+      auto res = close(i);
+      if (res == 0) {
+        std::cerr << "pid: " << getpid() << "successfully closed " << i << std::endl;
```
do we want this printout, or is it just a test?

ktf replied:
This was just a test. I will drop it.

ktf committed:
DPL: multiplex children stdout and stderr

This allows the driver to handle more children. A better solution would
be to completely remap stderr and stdout so that they write messages
over websocket.

ktf commented:
Tested also on hyperloop. Merging.
